backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts

```diff
@@ -236,7 +242,106 @@ export class ParcelDescriptionsService {
     parcelDescriptions: ParcelDescriptionInputDTO[],
     userInfo: any,
   ) {
-    // TODO: Implementation to come in another pull request
+    this.sitesLogger.log(
+      'parcelDescriptionService.addParcelDescriptionsForSite() start',
+    );
+    this.sitesLogger.debug(
+      'parcelDescriptionService.addParcelDescriptionsForSite() start',
+    );
+
+    const now = new Date();
+    // Parse the parcel description into a subdivision database entity.
+    const subdivisions: Subdivisions[] = parcelDescriptions.map(
+      (parcelDescription) => {
+        const pid =
+          parcelDescription.descriptionType === 'Parcel ID'
+            ? parcelDescription.idPinNumber
+            : null;
+        const pin =
+          parcelDescription.descriptionType === 'Crown Land PIN'
+            ? parcelDescription.idPinNumber
+            : null;
+        const crownLandsFileNo =
+          parcelDescription.descriptionType === 'Crown Land File Number'
+            ? parcelDescription.idPinNumber
+            : null;
+        return {
+          pid: pid,
+          pin: pin,
+          crownLandsFileNo: crownLandsFileNo,
+          dateNoted: parcelDescription.dateNoted,
+          srAction: parcelDescription.srAction,
+          userAction: parcelDescription.userAction,
+          whoUpdated: userInfo?.givenName,
+          whenUpdated: now,
+          whoCreated: userInfo?.givenName,
+          whenCreated: now,
+          pidStatusCd: 'N', // TODO: Determine what this is and set appropriately.
+        } as Subdivisions;
+      },
+    );
+
+    // Insert the new subdivisions into the database.
+    let insertResult: InsertResult;
+    try {
+      // There is a bug with TypeORM where it will only return columns for the
+      // first parent entity of an entity that uses inheritence. Here I'm using
+      // the querybuilder to work around it.
+      insertResult = await this.subdivisionsRepository
+        .createQueryBuilder()
+        .insert()
+        .into(Subdivisions)
+        .values(subdivisions)
+        .returning('*')
+        .execute();
+    } catch (error) {
+      this.sitesLogger.error(
+        'Exception occured in parcelDescriptionService.addParcelDescriptionsForSite() end',
+        JSON.stringify(error),
+      );
+      throw error;
```

Comment:
if possible can we have a proper formatted Exception here?  for reference disclosure service

Reply:
Sure: do you have a recommendation for which type of exception I should throw if not the TypeORM exception?

Reply:
I think HttpStatus.UNPROCESSABLE_ENTITY could be the exception, but I recommend to have a look on the SiteResolver and SiteService for further reference